src/js/signin/components/Signin.jsx

```diff
@@ -0,0 +1,152 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class Signin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleLogin = this.handleLogin.bind(this);
+    this.handleSignup = this.handleSignup.bind(this);
+  }
+
+  componentDidMount() {
+    this.checkLoggedInStatus();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.checkLoggedInStatus(nextProps.currentlyLoggedIn);
+  }
+
+  checkLoggedInStatus(loggedIn) {
+    if (this.props.currentlyLoggedIn || loggedIn) {
+      const nextParams = new URLSearchParams(window.location.search);
+      const nextPath = nextParams.get('next');
```

Comment:
Should this be checked in the Verify component as well? Or perhaps move it up one level to the Main component?

Reply:
Potentially—I'm not 100% clear yet on how the verify component will work. This check will not work as it currently stands because the user must be logged in to get to the Verification stage